### DISCUSSION TITLE
Create an selfhosted alternative GitHub Codespaces

### DISCUSSION THREAD

**Discussion by dedrisproject** [coder/code-server#6258] [General]
The context is - given a group of junior developers, I want to provide an area within the company where they can access with their own credentials, find the list of projects they are assigned to, and the tasks to be performed on that project. Then, there should be a button next to each project saying "Access workspace," and they will find VSCode Web.

The topic is:

How can we structure the Docker setup to ensure that upon initialization:
1. It receives the password via a FormPOST to guarantee access only if accessed through a password-protected screen.

2. How can I make the associated repository file already available so that the developer can have a ready-to-use environment?

**Comment by code-asher** [comment]
You might find https://github.com/coder/coder useful although it only manages credentials and provisions workspaces, it does not have the project/task management functionality you also mentioned.

**Reply by dedrisproject** [reply]
hi, i Know,  i will attach the codeserver to my project/task manager (IDP, internal developer platform) - i need only make isolated the workspace for single developer - i run multiple docker, 1 docker for workspace, but if a developer share the link to other, other can access to code server and see file - i need build a session with autologin, if developer share the link the system block the access to codespace to unautorized user

**Reply by code-asher** [reply]
I see, I think that is pretty much the use case for coder/coder.

If you have to roll your own then maybe [oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy) could be used to handle the authentication; have them log in with their GitHub account or something like that.

**Reply by richardokonicha** [reply]
Hello, I'm doing something similar, would it be possible to use code-server for this. 
Can I pass a git repository url using the folder/workspace query parameters to open the code on code-server or is this only possible with Coder and by provisioning new editors for every individual user/workspace.

**Reply by code-asher** [reply]
There is not currently a way to automatically clone a git repository with a query parameter with code-server.

With Coder the way we do it is by adding a `git clone` to the startup script in our template, and each project has their own template.  You could also add an "Open in Coder" button to each repository to streamline things.  https://coder.com/docs/v2/latest/templates/open-in-coder

**Reply by code-asher** [reply]
Actually you could probably have a single template and make the repository a variable rather than making templates for each project.  But you might want separate projects if the dependencies for each project are different.
